File: noteflow-web/src/types/note.ts
export interface TagResponseDto {
  id: string;
  name: string;
}

export interface NoteResponseDto {
  id: string;
  title: string | null;
  content: string;
  status: string;
  createdAt: string;
  updatedAt: string;
  tags: TagResponseDto[];
}

export interface NoteCreateRequest {
  title?: string;
  content: string;
  tagIds?: string[];
}

export interface NoteUpdateRequest {
  title?: string;
  content: string;
}

File: NoteFlowApi/NoteFlowApi/Models/Note.cs
using System.ComponentModel.DataAnnotations;

namespace NoteFlowApi.Models {

    public class Note {
        [Key]
        public Guid Id { get; set; } = Guid.NewGuid();

        [MaxLength(255)]
        public string? Title { get; set; }

        [Required]
        public string Content { get; set; } = string.Empty;

        [Required]
        public NoteStatus Status { get; set; } = NoteStatus.Active;

        [Required]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        [Required]
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        // Preparação para Auth — FK sem navegação por enquanto
        [Required]
        public Guid UserId { get; set; }

        // Navegação N:N via NoteTag
        public ICollection<NoteTag> NoteTags { get; set; } = [];
    }
}